DDF-2168: Add integration tests for resource download & caching

garrettfreibott opened the pull request:
#### What does this PR do?
Adds integration tests for federated resource downloading and caching. Also adds a reusable restito stub server class and a retrieval action class for live csw itest federation.
#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)?
@tbatie 
@oconnormi 
@bantillo 
@Lambeaux 
@cmthom10 
#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@figliold
@lessarderic

  - Added integration tests to test current resource downloading functionality.
  - Added reusable classes for setting up a federated CSW stub server for
  integration tests

lessarderic commented on distribution/test/itests/test-itests-common/src/test/java/ddf/common/test/restito/FederatedCswServer.java:
```diff
@@ -0,0 +1,205 @@
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.common.test.restito;
```
Wonder if we should move this class to a different package. Not sure people will think to look here for a mock federated source.

garrettfreibott replied:
Not sure where a better place would be though.

tbatie replied:
I'd call it ddf.common.test.mock.csw

garrettfreibott replied:
The comment at the top of that class actually mentions that it's a workaround for a commit that @ryeats contributed to restito, which should be in version 0.8.2. Maybe we should just upgrade to that version instead?

lessarderic commented on distribution/test/itests/test-itests-common/src/test/java/ddf/common/test/restito/FederatedCswServer.java:
```diff
@@ -0,0 +1,205 @@
+ * Encompasses a Restito CSW Stub Server for use in integration testing. Handles handshake responses
+ * and federated query responses for a csw endpoint.
+ */
+public class FederatedCswServer {
```
Should we add `Mock` or `Stub` somewhere in the name?

garrettfreibott replied:
I was considering naming it "MockCswStubServer". Although it's used for federation, having `Federated` in it's name doesn't seem correct since it's not technically "federated" until you connect it as a federated source. The class itself is just a stub server.

lessarderic replied:
`FederatedCswMockServer` sounds good to me.

lessarderic commented on distribution/test/itests/test-itests-common/src/test/java/ddf/common/test/restito/FederatedCswServer.java:
```diff
@@ -0,0 +1,205 @@
+     * @param httpRoot httpRoot address. Typically the non-secure root of "http://localhost:".
+     * @param port     Port number on which to respond.
+     */
+    public FederatedCswServer(String sourceId, String httpRoot, int port) {
```
Maybe we should change the constructor to take in a `DynamicUrl` and add a `getRoot()` and `getPort()` method to `DynamicUrl`? Thoughts?

tbatie replied:
I agree with this, mostly to enforce support for concurrent builds running and support for parallel itests.

garrettfreibott replied:
Unfortunately, DynamicUrl is in AbstractIntegrationTest and I would need to add a dependency on test-itest-ddf.

lessarderic commented on distribution/test/itests/test-itests-common/src/test/java/ddf/common/test/restito/FederatedCswServer.java:
```diff
@@ -0,0 +1,205 @@
+     * @param resourceName name of capability file
+     * @throws IOException
+     */
+    private void setCapabilityResponse(String resourceName) throws IOException {
```
`public`?

lessarderic commented on distribution/test/itests/test-itests-common/src/test/java/ddf/common/test/restito/FederatedCswServer.java:
```diff
@@ -0,0 +1,205 @@
+    }
+
+    private void setDefaultCapabilityResponse() throws IOException {
+        try (InputStream inputStream = getClass().getResourceAsStream(
```
Shouldn't this just delegate to `setCapabilityResponse()`?

garrettfreibott replied:
Unfortunately DynamicUrl is in AbstractIntegrationTest, which is in test-itest-ddf.

lessarderic commented on distribution/test/itests/test-itests-common/src/test/java/ddf/common/test/restito/FederatedCswServer.java:
```diff
@@ -0,0 +1,205 @@
+    }
+
+    private void setDefaultQueryResponse() throws IOException {
+        try (InputStream inputStream = getClass().getResourceAsStream("/csw-query-response.xml")) {
```
Delegate to `setQueryResponse()`?

lessarderic commented on distribution/test/itests/test-itests-common/src/test/java/ddf/common/test/restito/FederatedCswServer.java:
```diff
@@ -0,0 +1,205 @@
+        }
+    }
+
+    private String substituteTags(String document) {
```
I shouldn't have written this code and used `Library.getFileContent()` instead. Can you replace those methods?

garrettfreibott replied:
Library.getFileContent is inside test-itest-ddf as well :/

lessarderic commented on distribution/test/itests/test-itests-common/src/test/java/ddf/common/test/restito/RestitoRetrievalResponse.java:
```diff
@@ -0,0 +1,172 @@
+ * Allows for the simple setup of specific responses, slow sending, and simulated
+ * network failure retry testing.
+ */
+public class RestitoRetrievalResponse implements Function<Response, Response> {
```
How about `ChunkedContent` instead? That would be in line with the other Restito actions such as `resourceContent` and `stringContent`.

I was also thinking, maybe we should make the constructors `private` and have static factory methods like `Action chunkedContent(String responseMessage)`, `Action chunkedContent(String responseMessage, int messageDelayMs)`, etc., to replace the current `action()` method. With static imports, people would be able to use this like any of the other Restito Actions. Thoughts?

lessarderic replied:
Doing this would make it impossible to use `getCallCount()`, `resetCallCount()` and `resetCallStates()`, but I wonder if we absolutely need them. `getCallCount()` can be replaced with `verifyHttp()` I believe and the only scenario where the other two would be useful is if the same exact endpoint is called explicitly more than once and the same exact retries and failures are needed on all of those.

On top of that, it looks like when the next major Restito release comes out, it will support custom `Applicable` classes and chained `then`s, which would make those methods unnecessary.

garrettfreibott replied:
When we were creating the action() method, I was considering switching this class to be a static factory class instead of returning `this`.
I'm currently going through the tests and seeing if verifyHttp's times() method will cover what we need. The only problem is that the calls cannot be reset, so the number of calls would have to very specifically be monitored if a test were to have multiple call number assertions. If it works, then I'll convert this to a factory class instead of implementing `Function<Response, Response>`

lessarderic commented on distribution/test/itests/test-itests-common/src/test/java/ddf/common/test/restito/RestitoRetrievalResponse.java:
```diff
@@ -0,0 +1,172 @@
+     * @param responseMessage Message to be sent in response.
+     * @param messageDelayMs  Delay in ms to wait between sending each character of the message.
+     */
+    public RestitoRetrievalResponse(String responseMessage, int messageDelayMs) {
```
We should use `Duration` here instead of `int`, that way people can use `Duration.of(10, MILLIS)`. Or we could use `int time, TimeUnit unit` and have people do `10, MILLISECONDS`.

Wonder if we should also have a way to provide a number of chunks. That would allow people to send more than one character at a time. Thoughts?

garrettfreibott replied:
I switched it over to use a Duration object.

I don't think we should support having a chunk size. This class is starting to go a bit overboard. It does what we need it to do to support download testing. If in the future chunk size is needed, it should be straightforward enough to add another factory method to this class attach an Action that sends specified chunk sizes, but as of right now, I don't think we really need that level of specificity.

lessarderic commented on distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java:
```diff
@@ -151,9 +179,22 @@ public void beforeExam() throws Exception {
+            CswSourceProperties cswStubServerProperties =
+                    new CswSourceProperties(CSW_STUB_SOURCE_ID);
+            cswStubServerProperties.put("cswUrl", CSW_STUB_SERVER_PATH.getUrl());
+            cswStubServerProperties.put("pollInterval", 1);
```
We can remove this since we're using the same value.

lessarderic commented on distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java:
```diff
@@ -168,13 +209,17 @@ public void beforeExam() throws Exception {
             getServiceManager().createManagedService(CswSourceProperties.GMD_FACTORY_PID,
                     gmdProperties);
 
+            LOGGER.error("#### Starting wait for CSW stub server");
```
We can remove those `LOGGER.error()` calls.

lessarderic commented on distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java:
```diff
@@ -951,6 +1002,163 @@ public void testCswCreateEventEndpoint() throws Exception {
+    @Test
+    public void testDownloadFromFederatedCswSource() throws Exception {
+        RestitoRetrievalResponse restitoRetrievalResponse =
+                new RestitoRetrievalResponse("testData");
```
We probably should use variables instead of hard-coding `"testData"` in the test methods.

lessarderic commented on distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java:
```diff
@@ -951,6 +1002,163 @@ public void testCswCreateEventEndpoint() throws Exception {
+        // @formatter:off
+        when().get(restUrl).then().log().all().assertThat().contentType("text/plain")
+                .body(is("testData"));
+        // @formatter:on
```
I'm assuming this will trigger retries so we should use `verifyHttp()` here to make sure the endpoint has been called multiple times.

garrettfreibott replied:
This one should only be called once, it's not sent with a long delay or with any failures.

lessarderic replied:
I thought that a 2 second delay would cause retries? If not, do we have a test that's slow enough to cause retries?

garrettfreibott replied:
Sorry, github is acting up for me. It's been attaching my comments to the wrong threads and this comment thread is not even inside to the `testSlowResponse()` method you're referring to.

I'm not sure why the 2000 ms test does not cause a failure. The retry logic inside the downloader seems to infer that it will time out if it's longer than 1 second between receiving data. If I set the response to something high like 10000ms, it does fail. I'll look into it a bit.

garrettfreibott replied:
Looked into it a bit more. After not receiving anything for 6 seconds, our Future<ReliableResourceStatus> throws a cancellation exception and then waits 10 seconds before reattempting the download. I decided to remove this test altogether as it does not seem to be testing anything useful that the other tests aren't already testing.

lessarderic commented on distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java:
```diff
@@ -951,6 +1002,163 @@ public void testCswCreateEventEndpoint() throws Exception {
+        // @formatter:on
+    }
+
+    // Tests that if the endpoint disconnects twice, the retrieval retries both times
```
I'd use a JavaDoc comment `/** */` here.

garrettfreibott replied:
Also added javadocs to the other added tests to explain what they are testing specifically

lessarderic commented on distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java:
```diff
@@ -951,6 +1002,163 @@ public void testCswCreateEventEndpoint() throws Exception {
+        // @formatter:off
+        when().get(restUrl).then().log().all().assertThat().contentType("text/plain")
+                .body(is("testData"));
+        // @formatter:on
```
Use `verifyHttp()` here.

oconnormi replied:
hmm, i never touched this class...

lessarderic commented on distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java:
```diff
@@ -951,6 +1002,163 @@ public void testCswCreateEventEndpoint() throws Exception {
+        // @formatter:on
+    }
+
+    // Tests that is the endpoint disconnects 3 times, the retrieval fails after 3 attempts
```
JavaDoc comment.

lessarderic commented on distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java:
```diff
@@ -951,6 +1002,163 @@ public void testCswCreateEventEndpoint() throws Exception {
+        // @formatter:off
+        when().get(restUrl).then().log().all().assertThat().statusCode(500).contentType("text/plain")
+                .body(containsString("cannot retrieve product"));
+        // @formatter:on
```
Use `verifyHttp()`.

garrettfreibott replied:
What should I be verifying here? This is only testing that the ReliableResourceDownloader only retries 3 times before failing - as proven by passing the previous test which has 2 planned disconnects that this test which has 3.

lessarderic replied:
Good point. Ignore my comment.

lessarderic commented on distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java:
```diff
@@ -951,6 +1002,163 @@ public void testCswCreateEventEndpoint() throws Exception {
+        when().get(restUrl).then().log().all().assertThat().contentType("text/plain")
+                .body(is("testData"));
+        // @formatter:on
+        assertThat(restitoRetrievalResponse.getCallCount(), equalTo(1));
```
We should use `verifyHttp()` instead, especially if we remove `getCallCounts()` :-)

lessarderic commented on distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java:
```diff
@@ -951,6 +1002,163 @@ public void testCswCreateEventEndpoint() throws Exception {
+                        + "?transform=resource";
+
+        // Download product twice, should only call the stub server to download once
+        restitoRetrievalResponse.resetCallCount();
```
Any reason we need to call this here?

garrettfreibott replied:
It's not entirely necessary since the calls *should* be at 0 at this point. I decided to stick with the `reset counter -> make calls -> assert counter` order since that isolates the next assertion to the calls immediately following the reset.

lessarderic commented on distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java:
```diff
@@ -951,6 +1002,163 @@ public void testCswCreateEventEndpoint() throws Exception {
+                        + "?transform=resource";
+
+        // Download product twice, and change metacard on stub server between calls.
+        restitoRetrievalResponse.resetCallCount();
```
Same comments/questions as above.

lessarderic commented on distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java:
```diff
@@ -951,6 +1002,163 @@ public void testCswCreateEventEndpoint() throws Exception {
+        // @formatter:off
+        when().get(restUrl).then().log().all().assertThat().contentType("text/plain")
+                .body(is("testData"));
+        cswServer.setQueryResponse("csw-query-response-modified.xml");
```
Do you think we could use the same response file, have a replacement for the modified time and use `Library.getFileContent()` to replace it?

garrettfreibott replied:
I'd rather keep this method to allow for the entire response to be replaced.

lessarderic commented on distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java:
```diff
@@ -951,6 +1002,163 @@ public void testCswCreateEventEndpoint() throws Exception {
+        assertThat(restitoRetrievalResponse.getCallCount(), equalTo(2));
+    }
+
+    private void setupCswProductRetrieval() throws Exception {
```
Do we still need this?

garrettfreibott replied:
Removed

lessarderic commented on distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java:
```diff
@@ -1048,10 +1256,20 @@ private String ingestXmlWithProduct(String fileName) throws IOException {
     }
 
+    @Override
+    protected Option[] configureCustom() {
```
We can revert this change since we're not using `httpclient` anymore.

tbatie commented on distribution/test/itests/test-itests-common/pom.xml:
```diff
@@ -103,6 +103,17 @@
             <version>${project.version}</version>
         </dependency>
+        <dependency>
+            <groupId>com.xebialabs.restito</groupId>
```
Should we be using our ddf.test.thirdparty:restito instead?

garrettfreibott replied:
The comment at the top of that class actually mentions that it's a workaround for a commit that @ryeats contributed to restito, which should be in version 0.8.2. Maybe we should just upgrade to that version instead? I'll make a ticket to address updating our restito version to 0.8.2 and removing the ddf.test.thirdparty:Restito class. It would also probably be best if we had a restito variable since it's version is declared specifically in a few places.

tbatie commented on distribution/test/itests/test-itests-common/src/test/java/ddf/common/test/restito/FederatedCswServer.java:
```diff
@@ -0,0 +1,205 @@
+
+    private final int port;
+
+    private StubServer cswStubServer;
```
Might want to add a getter for the server for any customization this class doesn't support

lessarderic replied:
Shouldn't be needed since the class already provides the `whenHttp` and `verifyHttp` methods.

garrettfreibott replied:
The methods we have should be enough for testing, but I went ahead and added `getServer` since there are a few other public methods on StubServer has that may be needed for some reason.

garrettfreibott commented:
Addressed comments - Sorry about any comments that were left on the wrong comment thread. Github was acting up for me this morning and apparently scattered my replies all over. Should be cleaned up now though.

lessarderic commented on distribution/test/itests/test-itests-common/src/test/java/ddf/common/test/mock/csw/ChunkedContent.java:
```diff
@@ -0,0 +1,158 @@
+     * This should be called after any complete failure tests to ensure that the response will not
+     * continue waiting for another retry attempt from the calling server.
+     */
+    public void resetFailState() {
```
Don't think this can be used anymore now that we're using static factory methods that return `Action` objects. We should remove it.

lessarderic commented on distribution/test/itests/test-itests-common/src/test/java/ddf/common/test/mock/csw/ChunkedContent.java:
```diff
@@ -0,0 +1,158 @@
+     *
+     * @return the minimum headers required for sending a chunked data response
+     */
+    public static Action getHeaders() {
```
This should probably be `private`. Don't think there's any reason someone should use this on its own.

lessarderic commented on distribution/test/itests/test-itests-common/src/test/java/ddf/common/test/mock/csw/ChunkedContent.java:
```diff
@@ -0,0 +1,158 @@
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.common.test.mock.csw;
```
This class doesn't contain anything related to CSW so it probably should go in the `ddf.common.test.restito` package in the `test-itests-common` module.

lessarderic commented on distribution/test/itests/test-itests-common/src/test/java/ddf/common/test/mock/csw/ChunkedContent.java:
```diff
@@ -0,0 +1,158 @@
+     *                         the first character. Once this number is reached, the message will
+     *                         send successfully.
+     */
+    public static Action newResponse(String responseMessage, Duration messageDelay,
```
As I mentioned in my previous comments, I would try to stay as close to the `Action` naming convention as possible and call these methods `chunkedContent`. I know it looks a little redundant, but it looks better if you use static imports. Here's an example:

```
import static ddf.common.test.restito.ChunkedResponse.chunkedResponse;

@Test
public void test() {
    whenHttp(server).match(...).then(chunkedContent(...));
}
```

lessarderic commented on distribution/test/itests/test-itests-common/src/test/java/ddf/common/test/mock/csw/ChunkedContent.java:
```diff
@@ -0,0 +1,158 @@
+     * @return
+     */
+    @Override
+    public Response apply(Response response) {
```
Something to consider. Maybe we should have a `private` inner-class implement the `Function<>` interface instead of having `ChunkedContent` implement it. That way, the `ChunkedContent` class wouldn't need to have a `public` `apply` method and would only contain the `static` factory methods which would make a better API.

garrettfreibott replied:
Definitely

lessarderic commented on distribution/test/itests/test-itests-common/src/test/java/ddf/common/test/mock/csw/FederatedCswMockServer.java:
```diff
@@ -0,0 +1,207 @@
+     * Constructor for the federated CSW Restito stub server.
+     *
+     * @param sourceId DDF sourceId to give to the stub server.
+     * @param httpRoot httpRoot address. Typically the non-secure root of "http://localhost:".
```
Should we remove the `of` from the second sentence?

lessarderic commented on distribution/test/itests/test-itests-common/src/test/java/ddf/common/test/mock/csw/FederatedCswMockServer.java:
```diff
@@ -0,0 +1,207 @@
+    }
+
+    private String substitutePortNumber(String document) {
+        return document.replaceAll("\\{\\{port\\}\\}", Integer.toString(port));
```
Since we can't use `Library.getFileContent()` :-( we should at least be consistent and use the same replacement strings, i.e., `$port$`, `$sourceId$` and `$httpRoot$`.

Lambeaux commented on distribution/test/itests/test-itests-common/src/test/java/ddf/common/test/mock/csw/FederatedCswMockServer.java:
```diff
@@ -0,0 +1,207 @@
+     * @param port     Port number on which to respond.
+     */
+    public FederatedCswMockServer(String sourceId, String httpRoot, int port) {
+        if (httpRoot.endsWith(":")) {
```
You'd think there would be facilities for handling cases like this without us having to resort to it ourselves.

Lambeaux commented on distribution/test/itests/test-itests-common/src/test/java/ddf/common/test/mock/csw/FederatedCswMockServer.java:
```diff
@@ -0,0 +1,207 @@
+            setDefaultCapabilityResponse();
+            setDefaultQueryResponse();
+        } catch (IOException | RuntimeException e) {
+            fail("Failed to setup " + getClass().getSimpleName() + ": " + e.getMessage());
```
I don't usually care about the small stuff, but @lessarderic is this better off with `String.format`?

lessarderic replied:
👍

Lambeaux commented on distribution/test/itests/test-itests-common/src/test/java/ddf/common/test/mock/csw/FederatedCswMockServer.java:
```diff
@@ -0,0 +1,207 @@
+     * Returns VerifyHttp's verifyHttp method with the stub server as the parameter. Use is the same.
+     * Provides itest verification when a specific Http call is made.
+     *
+     * @return VerifyHttp using verifyHttp method with the tub server as the parameter.
```
Is the server taking a bath?

vinamartin replied:
I thought you were just being cute here, but you caught a typo too. :) Nice.

Lambeaux commented on distribution/test/itests/test-itests-common/src/test/java/ddf/common/test/mock/csw/FederatedCswMockServer.java:
```diff
@@ -0,0 +1,207 @@
+        try (InputStream inputStream = getClass().getResourceAsStream("/" + resourceName)) {
+            String document = substituteTags(IOUtils.toString(inputStream));
+
+            LOGGER.debug("Capability response: \n{}", document);
```
That might break on Windows. Use `String.format` and the `%n` special symbol, which will be replaced with the proper terminator.

rzwiefel replied:
Using the string interpolation we get from slf4j is fine. It would not break, just not render a new line (even though I thought Java took care of that for you). Either way it's just a debug statement.

rzwiefel replied:
If anything just add another interpolated part 

```java
LOGGER.debug("Capability response: {}{}", System.lineSeperator(), document);
```

rzwiefel replied:
I can't spell. Also, I'd say it's even worth of a static import just to be succinct.

garrettfreibott replied:
Not entirely sure if this is an issue with slf4j. Doing a quick search shows `\n` used extensively in our logging statements in DDF. Worst case, the first line of the document won't be on it's own line on windows machines - which in this case is just xml schema stuff.

Lambeaux commented on distribution/test/itests/test-itests-common/src/test/java/ddf/common/test/mock/csw/FederatedCswMockServer.java:
```diff
@@ -0,0 +1,207 @@
+        try (InputStream inputStream = getClass().getResourceAsStream("/" + resourceName)) {
+            String document = substituteTags(IOUtils.toString(inputStream));
+
+            LOGGER.debug("Query response: \n{}", document);
```
Same issue as above.

garrettfreibott replied:
Same comment as above.

Lambeaux commented on distribution/test/itests/test-itests-common/src/test/java/ddf/common/test/restito/ChunkedContent.java:
```diff
@@ -0,0 +1,154 @@
+    }
+
+    /**
+     * Constructor for a response that has a delay and a number of planned failure.
```
failure --> failures

Lambeaux commented on distribution/test/itests/test-itests-common/src/test/java/ddf/common/test/restito/ChunkedContent.java:
```diff
@@ -0,0 +1,154 @@
+    }
+
+    /**
+     * Returns an composite action that holds the headers required for a plain text response as well
```
`Returns *a* composite action ... `

Lambeaux commented on distribution/test/itests/test-itests-common/src/test/java/ddf/common/test/restito/ChunkedContent.java:
```diff
@@ -0,0 +1,154 @@
+        private int numberOfRetries = 0;
+
+        /**
+         * Constructor for a response that has a delay and a number of planned failure.
```
failure --> failures

Lambeaux commented on distribution/test/itests/test-itests-common/src/test/java/ddf/common/test/restito/ChunkedContent.java:
```diff
@@ -0,0 +1,154 @@
+                        numberOfRetries++;
+                        break;
+                    }
+                } catch (IOException | InterruptedException e) {
```
Best practice for catching an `InterruptedException` is to also do the following:
`Thread.currentThread().interrupt();`

rzwiefel replied:
1) It would probably be better to actually deal with it, if anything. (eg, retry the write, etc)
2) This is for testing, and probably has no benefit being reinterrupted in this case since we aren't constructing or adhearing to _flexible cancellation policies_ or the like.

garrettfreibott replied:
It's a test class. If it fails to write, it fails the test.

Lambeaux commented on distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java:
```diff
@@ -852,11 +899,11 @@ public void testCswDurableSubscription() throws Exception {
         bundle.start();
         while (bundle.getState() != Bundle.ACTIVE) {
-            Thread.sleep(1000);
+            sleep(1000);
```
Ensure the catch block for the interrupted exception sets the flag.

garrettfreibott replied:
This line is a just a reformat. This test doesn't use the stub server or a ChunkedContent object that contains the Interrupted catch.

Lambeaux commented on distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java:
```diff
@@ -964,7 +1130,7 @@ private void verifyEvents(Set<String> metacardIdsExpected, Set<String> metacardI
 
             try {
-                Thread.sleep(EVENT_UPDATE_WAIT_INTERVAL);
+                sleep(EVENT_UPDATE_WAIT_INTERVAL);
```
Handle interrupt.

Lambeaux commented on distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java:
```diff
@@ -964,7 +1130,7 @@ private void verifyEvents(Set<String> metacardIdsExpected, Set<String> metacardI
-                Thread.sleep(EVENT_UPDATE_WAIT_INTERVAL);
+                sleep(EVENT_UPDATE_WAIT_INTERVAL);
                 millis += EVENT_UPDATE_WAIT_INTERVAL;
             } catch (InterruptedException e) {
```
`Thread.currentThread().interrupt();`

garrettfreibott replied:
This is another line reformat and is not used in any of the added tests.

Lambeaux commented:
So I feel bad for catching a lot of grammar and spelling errors. Price of comments I suppose. Fix at your discretion. I say the core of this LGTM 👍

rzwiefel commented:
:+1:

tbatie commented:
Did a quick build and ran TestFederation. Passed successfully. :+1:
